client/my-sites/exporter/export-card/index.jsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { localize } from 'i18n-calypso';
+import flowRight from 'lodash/flowRight';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SpinnerButton from './spinner-button';
+import FoldableCard from 'components/foldable-card';
+import analytics from 'lib/analytics';
+import Interval, { EVERY_SECOND } from 'lib/interval';
+import AdvancedSettings from './advanced-settings';
+import {
+	advancedSettingsFetch,
+	exportStatusFetch,
+	setPostType,
+	startExport,
+} from 'state/site-settings/exporter/actions';
+import {
+	shouldShowProgress,
+	getSelectedPostType,
+	isExporting,
+} from 'state/site-settings/exporter/selectors';
+
+class ExportCard extends Component {
+	componentWillMount() {
+		this.props.advancedSettingsFetch( this.props.siteId );
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.siteId !== this.props.siteId ) {
+			this.props.advancedSettingsFetch( newProps.siteId );
+		}
+	}
+
+	trackExportClick = ( scope = 'all' ) => {
+		analytics.tracks.recordEvent( 'calypso_export_start_button_click', { scope } );
```

Comment:
Further, the analytics Redux actions are composable over other actions. In this way, we can be even more explicit in separating the concerns from our components and their side-effects.

```js
const trackExportsClick = ( scope = 'all' ) => recordTracksEvent( 'calypso_export_start_button_click', { scope } );

const mapDispatchToProps = dispatch => ( {
	startExport: ( { exportAll } ) => dispatch( withAnalytics(
		trackExportsClick( exportAll ? 'all' : 'selected' ),
		startExport( { exportAll } )
	) )
} );
```

This way your React component doesn't even need to be aware of the fact that the clicks are being tracked. It can focus on what it _wants_ and _needs_ to know, which is whether or not to start and export and of which type.